maintenance_plan_meter_mode/models/maintenance.py

```diff
@@ -0,0 +1,129 @@
+        readonly=True)
+
+    @api.onchange('meter_mode')
+    def _onchange_meter_mode(self):
+        if self.meter_mode:
+            self.interval = None
+            self.interval_step = None
+            self.maintenance_plan_horizon = None
+            self.planning_step = None
+        else:
+            self.meter_current_value = None
+            self.meter_autoinc_value = None
+            self.meter_autoinc_unit = None
+            self.meter_autoinc_maxvalue = None
+            self.meter_autoinc_lastvalue = None
+            self.meter_autoinc_lastupdate = None
+
+    @api.constrains('meter_mode')
+    def _check_meter_mode(self):
+        for plan in self:
+            if plan.meter_mode:
+                if not plan.meter_autoinc_value or not plan.meter_autoinc_unit:
+                    raise ValidationError(
+                        _("It can´t be Self-increase mode"
+                          " without Value to self-increase per day"))
+                if not plan.meter_autoinc_maxvalue:
+                    raise ValidationError(
+                        _("It can´t be Self-increase mode"
+                          " without Maximum amount since the last revision"))
+
+    def _review_meter_mode_values(self, values):
+        meter_mode = values['meter_mode'] \
+            if 'meter_mode' in values else self.meter_mode
+        if meter_mode:
+            values['interval'] = None
+            values['interval_step'] = None
+            values['maintenance_plan_horizon'] = None
+            values['planning_step'] = None
+        else:
+            values['meter_current_value'] = None
+            values['meter_autoinc_value'] = None
+            values['meter_autoinc_unit'] = None
+            values['meter_autoinc_maxvalue'] = None
+            values['meter_autoinc_lastvalue'] = None
+            values['meter_autoinc_lastupdate'] = None
+        return values
+
+    @api.model
+    def create(self, values):
+        values = self._review_meter_mode_values(values)
+        return super(MaintenancePlan, self).create(values)
+
+    @api.multi
+    def write(self, values):
+        values = self._review_meter_mode_values(values)
+        return super(MaintenancePlan, self).write(values)
+
+
+class MaintenanceEquipment(models.Model):
+
+    _inherit = 'maintenance.equipment'
+
+    @api.model
+    def _cron_generate_requests(self):
+        # Generates maintenance request on the next_maintenance_date
+        for plan in self.env['maintenance.plan'].search(
+                [('meter_mode', '=', True), ('meter_autoinc_value', '>', 0)]):
+            equipment = plan.equipment_id
+            equipment._create_new_request_meter_mode(plan)
+
+    def _create_new_request_meter_mode(self, mp):
+        # Compute meter_current_value adding meter_autoinc_value from last day
+        if not mp.meter_autoinc_lastupdate:
+            mp.meter_autoinc_lastupdate = fields.Date.from_string(fields.Date.today())
```

Comment:
In v12 `from_string()` is no longer required, every reference like this one can be removed:
```suggestion            
            mp.meter_autoinc_lastupdate = fields.Date.today()
```

Reply:
Ok. I'll remove this.
I had seen that in maintenance_plan addon actually use fields.Date.from_string just in _create_new_request function.

Reply:
Yes, when OCA migrated `maintenance_plan` to v12 didn't update this part of code, but it should've be done IMO.